bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,284 @@
+    # TODO: Not used, maybe remove
+    def get_prev_depth_node(self):
+        for prev_item in reversed(self._state_stack):
+            if prev_item["depth"] == self.depth - 1:
+                return prev_item
+        return
 
+    def skip(self):
+        self._skipped.append(
+            {
+                "line": self.item["line"],
+                "column": self.item["line"],
+                "kind": self.kind,
+                "name": self.name,
+            }
+        )
 
-in_struct = False
-module_linelist = []
+    def close(self):
+        if self._state_stack[-1]["kind"] == "NAMESPACE":
+            self.linelist.append("}")
+        if self._state_stack[-1]["kind"] == "STRUCT_DECL":
+            self.linelist.append(";")
+        if self._state_stack[-1]["kind"] == "CLASS_TEMPLATE":
+            self.linelist.append(";")
 
+    def handle_node(self, item):
+        self.item = item
+        self.kind = self.item["kind"]
+        self.name = self.item["name"] if "name" in self.item else ""
+        self.members = self.item["members"]
+        self.depth = self.item["depth"]
 
-def handle_final(filename, module_name):
-    linelist = []
-    linelist.append(f"#include <{filename}>")
-    linelist.append("#include <pybind11/pybind11.h>")
-    linelist.append("namespace py = pybind11;")
-    for i in range(len(module_linelist)):
-        if module_linelist[i].startswith("namespace"):
-            continue
-        else:
-            module_linelist[i] = "".join(
-                (f"PYBIND11_MODULE({module_name}, m)", "{", module_linelist[i])
-            )
-            break
-    for line in module_linelist:
-        linelist.append(line)
-    linelist.append("}")
-    return linelist
+        self._state_stack.append(
+            {"kind": self.kind, "name": self.name, "depth": self.depth}
+        )
 
+        self.kind_functions[self.kind]()
 
-def handle_include(item):
-    pass
+        if self.kind_functions[self.kind] is not self.skip:
+            for sub_item in self.members:
+                self.handle_node(sub_item)
 
+        self.close()
 
-def handle_alias(item):
-    pass
+        self._state_stack.pop()
 
+    def handle_namespace(self):
+        self.linelist.append(f"namespace {self.name}" + "{")
 
-def handle_constructor(item):
-    parameters_type = ""
-    parameters_type = ",".join(params for params in item["parameters_type"])
-    module_linelist.append(f".def(py::init<{parameters_type}>())")
+    def get_fields_from_anonymous(self, item):
+        fields = []
+        for sub_item in item["members"]:
+            if sub_item["kind"] in ("ANONYMOUS_UNION_DECL", "ANONYMOUS_STRUCT_DECL"):
+                for field in self.get_fields_from_anonymous(sub_item):
+                    fields.append(field)
+            if sub_item["kind"] == "FIELD_DECL":
+                fields.append(sub_item)
+        return fields
 
+    def handle_struct_decl(self):
+        cxx_base_specifier_list = [
```

Comment:
Maybe rename to `base_class_list` or something similar if this is what I think it is. Else... document what this is